h/static/scripts/cross-frame-service.coffee

```diff
@@ -0,0 +1,72 @@
+# Instantiates all objects used for cross frame discovery and communication.
+class CrossFrameService
+  providers: null
+
+  this.inject = ['$rootScope', '$document', '$window', 'store', 'annotationUI']
+  constructor: ($rootScope, $document, $window, store, annotationUI) ->
+    @providers = []
+
+    createDiscovery = ->
+      options =
+        server: true
+      new CrossFrameDiscovery(options)
+
+    # Set up the bridge plugin, which bridges the main annotation methods
+    # between the host page and the panel widget.
+    createBridge = ->
+      options =
+        scope: 'annotator:bridge'
+      new CrossFrameBridge(options)
+
+    createAnnotationSync = (bridge) ->
+      whitelist = ['target', 'document', 'uri']
+      options =
+        formatter: (annotation) ->
+          formatted = {}
+          for k, v of annotation when k in whitelist
+            formatted[k] = v
+          formatted
+        parser: (annotation) ->
+          parsed = new store.AnnotationResource()
+          for k, v of annotation when k in whitelist
+            parsed[k] = v
+          parsed
+        emit: (args...) ->
+          $rootScope.$emit.call($rootScope, args...)
+        on: (event, handler) ->
+          $rootScope.$on(event, handler)
+
+      new AnnotationSync(options, bridge)
+
+    createAnnotationUISync = (bridge) ->
+      new AnnotationUISync($rootScope, $window, bridge, annotationUI)
+
+    addProvider = (channel) =>
```

Comment:
This whole function seems a bit weird. Why is this in here? Why is it not part of one or other of the `*Sync` objects?